### PR TITLE
First attempt to implement a definition to yahoo/dnscache

### PR DESCRIPTION
**Pull request by HuskyMoonMoon**
Please fill in this template.

- [/] Use a meaningful title for the pull request. Include the name of the package modified.
- [/] Test the change in your own code. (Compile and run.)
- [/] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [/] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [/] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [/] The package does not provide its own types, and you can not add them.
- [/] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [/] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [/] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

First attempt to implement a definition to `yahoo/dnscache`, mostly copied from `@types/node`'s `dns` because `dnscache` itself is wrapped around `dns`

### CODE REVIEW

File: types/dnscache/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    BADFLAGS: string;
+    NONAME: string;
+    BADHINTS: string;
+    NOTINITIALIZED: string;
+    LOADIPHLPAPI: string;
+    ADDRGETNETWORKPARAMS: string;
+    CANCELLED: string;
+}
+
+declare var dnscache: DnsCache;
```
Comment (armanio123) [inline types/dnscache/index.d.ts]: Use const instead of var.

File: types/dnscache/dnscache-tests.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import * as dnscache from "dnscache";
```
Comment (armanio123) [inline types/dnscache/dnscache-tests.ts]: Use require("module") as the d.ts is using commonjs modules.

File: types/dnscache/tslint.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "unified-signatures": false,
```
Comment (armanio123) [inline types/dnscache/tslint.json]: Don't disable lint rules.

File: types/dnscache/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+declare module "dnscache" {
```
Comment (armanio123) [inline types/dnscache/index.d.ts]: No need to declare module just use the "export =" instead.

**Review by armanio123 — CHANGES_REQUESTED** [review]

**Comment by armanio123** [comment]
The latest commit doesn't address the import x = require("module") syntax. Please fix that before approving the PR.

**Comment by minestarks** [comment]
@Nagato-Yuki please address the review comments to avoid the PR being closed due to inactivity. Thanks!

**Comment by paulvanbrenk** [comment]
And closing due to inactivity

**Pull request closed without merging**